### PR TITLE
[feature] Remove API Prefix

### PR DESCRIPTION
**Pull request by brianjgeiger**
Purpose
-----------
@icereval will be moving the API to its own subdomain, which means that the whole concept of an API Prefix will be nothing more than historical overhead. Rather than keeping that overhead and complicating things, we are removing it, making a sleeker, better API experience.

Changes
------------
Remove the API_PATH and API_PREFIX settings and ensuring that everything that use them no longer do so.

Side effects
----------------
api_v2_url is a little more generic than it needs to be. It could potentially be simplified a bit.

This is necessary to support #3243 but will not close the issue on its own.

### CODE REVIEW

File: api/base/utils.py [hunk 1]
```diff
@@ -16,8 +15,7 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     """Like django's `reverse`, except returns an absolute URL. Also add query parameters."""
     relative_url = reverse(view_name, kwargs=kwargs)
 
-    url = website_util.api_v2_url(relative_url, params=query_kwargs,
-                                  base_prefix=api_settings.API_PATH)
+    url = website_util.api_v2_url(relative_url, params=query_kwargs, base_prefix='')
```
Comment (abought) [inline api/base/utils.py]: Since the function call signature is being changed to use `base_prefix=''` as the default, there's no need for that argument here.

File: website/util/__init__.py [hunk 2]
```diff
@@ -58,7 +57,7 @@ def api_url_for(view_name, _absolute=False, _xml=False, *args, **kwargs):
 def api_v2_url(path_str,
                params=None,
                base_route=website_settings.API_DOMAIN,
-               base_prefix=api_settings.API_PREFIX,
+               base_prefix='',
```
Comment (abought) [inline website/util/__init__.py]: Perhaps it would be helpful for one of the unit tests to cover the new domain expected in deployment, with the new default value for base_prefix? (eg change `test_api_v2_url_with_params`)

**Comment by lyndsysimon** [comment]
No comments beyond what Andy has already added.

File: website/util/__init__.py (hunk 2 above)
Comment (brianjgeiger) [inline website/util/__init__.py]: Done

**Comment by sloria** [comment]
Looks good. :octopus: Once the build is passing, this is good to merge.

**Comment by brianjgeiger** [comment]
#3330 needs to be finished and merged before this gets loaded onto staging2.

File: website/util/__init__.py (hunk 2 above)
Comment (abought) [inline website/util/__init__.py]: In the process of throwing out API_PREFIX, we accidentally dropped `API_BASE` as well. (and since the tests were deployment-independent, it wasn't obvious)  

I think that was an accident (refer to `api/base/urls.py`) and will submit another PR to put that back.
